klasses/views.py

```diff
@@ -0,0 +1,76 @@
+"""
+Views for Klasses app
+"""
+import logging
+
+from django.contrib.auth.models import User
+from rest_framework import (
+    authentication,
+    permissions,
+    status,
+)
+from rest_framework.generics import get_object_or_404
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from backends.edxorg import EdxOrgOAuth2
+from klasses.api import get_user_klasses_info, get_user_klass_info
+from klasses.permissions import CanReadIfSelf
+
+log = logging.getLogger(__name__)
+
+
+class UserKlassList(APIView):
+    """
+    Class based view for user klasse list view.
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+    )
+    permission_classes = (
+        permissions.IsAuthenticated,
+        CanReadIfSelf,
+    )
+
+    def get(self, request, username, *args, **kargs):  # pylint: disable=unused-argument
+        """
+        Returns information about the payments for the user in all the klasses she is allowed to pay.
+        """
+        user = get_object_or_404(
+            User,
+            social_auth__uid=username,
+            social_auth__provider=EdxOrgOAuth2.name
+        )
+
+        return Response(get_user_klasses_info(user=user))
+
+
+class UserKlassDetail(APIView):
```

Comment:
you don't deal with the `Klass` object in this view as it's written, but my suggestion is that you _should_. the view logic right now suggests that the `klass_id` should fetch us a single `Klass` object, and if it doesn't match, we eventually return a 404. implementing the view in the way i suggested would do that automatically via the tools that DRF already gives us. the line to cast `klass_id` as an int (`klass_id = int(klass_id)`) and fetch the `Klass` based on the id in `get_user_klass_info` are unnecessary boilerplate, IMO. using those `lookup_` parameters is a recommended pattern in DRF, and i think this is a good use case for it